sdk/js/src/api/stream/shared.js

```diff
@@ -31,3 +30,35 @@ export const MESSAGE_TYPES = Object.freeze({
   PUBLISH: 'publish',
   ERROR: 'error',
 })
+
+export const INITIAL_LISTENERS = Object.freeze(
+  Object.values(EVENTS).reduce((acc, curr) => ({ ...acc, [curr]: {} }), {}),
+)
+
+export const newQueuedListeners = listeners => {
```

Comment:
The queuing is required in order to ensure that we still keep the events before `open`, and that we don't emit them before `open`.